src/platform/backends/libvirt/libvirt_virtual_machine.cpp

```diff
@@ -199,22 +199,33 @@ auto domain_by_definition_for(const mp::VirtualMachineDescription& desc, const s
     return domain;
 }
 
-auto domain_state_for(virDomainPtr domain)
+auto domain_state_for(const mp::VirtualMachine::State& current_instance_state, virDomainPtr domain)
 {
-    auto state{0};
+    auto domain_state{0};
 
-    if (!domain)
+    if (!domain || virDomainGetState(domain, &domain_state, nullptr, 0) == -1 || domain_state == VIR_DOMAIN_NOSTATE)
         return mp::VirtualMachine::State::unknown;
 
     if (virDomainHasManagedSaveImage(domain, 0) == 1)
         return mp::VirtualMachine::State::suspended;
 
-    if (virDomainGetState(domain, &state, nullptr, 0) == -1)
-    {
-        return mp::VirtualMachine::State::unknown;
-    }
+    if (domain_state == VIR_DOMAIN_SHUTOFF)
```

Comment:
Why don't you deal with the other possible states like SHUTDOWN, CRASHED, PAUSED & BLOCKED?

Reply:
I suppose I should account for those libvirt states as well.  I'll add them.

Reply:
Ok, I have accounted for all of the domain states and basically set them to `off` for the returned instance state.